cluster/addons/iso-client/opaque/opaque.go

```diff
@@ -61,20 +35,36 @@ func getNode() (string, error) {
 }
 
 // path for opaque resources
-func (opaque *OpaqueIntegerResourceAdvertiser) generateOpaqueResourcePath() string {
-	return fmt.Sprintf("/status/capacity/pod.alpha.kubernetes.io~1opaque-int-resource-%s", opaque.Name)
+func generateOpaqueResourcePath(name string) string {
+	return fmt.Sprintf("/status/capacity/pod.alpha.kubernetes.io~1opaque-int-resource-%s", name)
```

Comment:
The prefix is also available as a constant -- either way you go, better to only have one copy of that awful prefix string anywhere.